File: frontend/src/services/dashboard.service.ts
import api from './api';
import { ApiResponse } from '../types';

export const dashboardService = {
  getStats: async () => {
    const res = await api.get<ApiResponse<{
      totalTours: number; activeTours: number; totalBookings: number;
      pendingBookings: number; confirmedBookings: number;
      totalCustomers: number; totalReviews: number;
    }>>('/Dashboard/stats');
    return res.data;
  },
  getRevenue: async (params?: { fromDate?: string; toDate?: string }) => {
    const res = await api.get<ApiResponse<any>>('/Reports/revenue', { params });
    return res.data;
  },
  getTourReport: async (params?: { fromDate?: string; toDate?: string }) => {
    const res = await api.get<ApiResponse<any>>('/Reports/tours', { params });
    return res.data;
  },
  getBookingReport: async (params?: { fromDate?: string; toDate?: string }) => {
    const res = await api.get<ApiResponse<any>>('/Reports/bookings', { params });
    return res.data;
  },
};

File: be/Infrastructure/Services/DashboardService.cs
using Core.DTOs.Response;
using Core.Enums;
using Core.Interfaces;
using Microsoft.EntityFrameworkCore;
using Shared;

namespace Infrastructure.Services;

public class DashboardService : IDashboardService
{
    private readonly IUnitOfWork _unitOfWork;

    public DashboardService(IUnitOfWork unitOfWork)
    {
        _unitOfWork = unitOfWork;
    }

    public async Task<ApiResponse<DashboardStatsResponse>> GetStatsAsync()
    {
        var totalTours = await _unitOfWork.Tours.CountAsync();
        var activeTours = await _unitOfWork.Tours.CountAsync(t => t.IsActive);
        var totalBookings = await _unitOfWork.Bookings.CountAsync();
        var pendingBookings = await _unitOfWork.Bookings.CountAsync(b => b.Status == BookingStatus.Pending);
        var confirmedBookings = await _unitOfWork.Bookings.CountAsync(b => b.Status == BookingStatus.Confirmed);
        var totalCustomers = await _unitOfWork.Customers.CountAsync();
        var totalReviews = await _unitOfWork.Reviews.CountAsync();

        var stats = new DashboardStatsResponse
        {
            TotalTours = totalTours,
            ActiveTours = activeTours,
            TotalBookings = totalBookings,
            PendingBookings = pendingBookings,
            ConfirmedBookings = confirmedBookings,
            TotalCustomers = totalCustomers,
            TotalReviews = totalReviews
        };

        return ApiResponse<DashboardStatsResponse>.Ok(stats);
    }
}
